client/src/redux/save-challenge-saga.js

```diff
@@ -0,0 +1,41 @@
+import { call, takeEvery, put, select } from 'redux-saga/effects';
+import { postSaveChallenge } from '../utils/ajax';
+import {
+  challengeDataSelector,
+  challengeMetaSelector
+} from '../templates/Challenges/redux';
+import { saveChallengeComplete } from './';
+import { createFlashMessage } from '../components/Flash/redux';
+import { challengeTypes } from '../../utils/challenge-types';
+
+export function* saveChallengeSaga() {
+  const { id, challengeType } = yield select(challengeMetaSelector);
+  const { challengeFiles } = yield select(challengeDataSelector);
+
+  if (challengeType === challengeTypes.multiFileCertProject) {
+    try {
+      const response = yield call(postSaveChallenge, { id, challengeFiles });
```

Comment:
```suggestion
      const response = yield call(postSaveChallenge, { id, files: challengeFiles.reduce(
          (acc, { fileKey, ...curr }) => [...acc, { ...curr, key: fileKey }],
          []
        ) });
```
Wrangles the challengeFiles into files